### PR TITLE
Monaco with LS

### PR DESCRIPTION
**Pull request by hoathaivan**

### CODE REVIEW

File: cnext_app/lib/codemirror/autocomplete-lsp/ls-plugin.js [hunk 1]
```diff
@@ -378,6 +381,8 @@ class LanguageServerPlugin {
                     documentText: this.view.state.doc.toString(),
                 });
 
+                console.log("testttttttt", line, character, triggerKind, triggerCharacter);
```
Comment (mbui0327) [inline cnext_app/lib/codemirror/autocomplete-lsp/ls-plugin.js:384]: remove this pls

File: cnext_app/lib/components/code-panel/monaco/CodeEditor.tsx [hunk 2]
```diff
@@ -310,6 +312,21 @@ const CodeEditor = ({ stopMouseEvent }) => {
         }
     });
 
+    useEffect(() => {
+        if (monaco) {
+            const path = store.getState().projectManager.activeProject?.path;
+            const pyLanguageServer = {
+                serverUri: "ws://localhost:3001/python",
```
Comment (mbui0327) [inline cnext_app/lib/components/code-panel/monaco/CodeEditor.tsx:319]: is this port correct?

File: cnext_app/lib/components/code-panel/monaco/CodeEditor.tsx [hunk 3]
```diff
@@ -310,6 +312,21 @@ const CodeEditor = ({ stopMouseEvent }) => {
         }
     });
 
+    useEffect(() => {
+        if (monaco) {
+            const path = store.getState().projectManager.activeProject?.path;
```
Comment (mbui0327) [inline cnext_app/lib/components/code-panel/monaco/CodeEditor.tsx:317]: I don't think this will be called when activeProject changed.

Comment (hoathaivan) [reply cnext_app/lib/components/code-panel/monaco/CodeEditor.tsx:317]: the Editor will re-render again if we change the activate the project right?
I was tested with the same name of the file. It still working well @mbui0327

File: cnext_app/lib/components/code-panel/monaco/CodeEditor.tsx (hunk 2 above)
Comment (hoathaivan) [reply cnext_app/lib/components/code-panel/monaco/CodeEditor.tsx:319]: corrected

**Pull request closed without merging**